pkg/asset/manifests/clusterapi/cluster.go

```diff
@@ -104,38 +106,44 @@ func (c *Cluster) Generate(dependencies asset.Parents) error {
 		if err != nil {
 			return errors.Wrap(err, "unable to inject installation info")
 		}
+
+		masterSecret := corev1.Secret{
```

Comment:
Nit, typically you make this a pointer since `runtime.Object` and various other interfaces expect this to be a pointer, not sure it matters here so long as the GVK is being set. 

Are there any tests you can modify to make sure that the GVK is actually being set?

```suggestion
		masterSecret := &corev1.Secret{
```

Reply:
This is being tested by "do not merge" commit on our CAPO series, [`b284ecd` (#7839)](https://github.com/openshift/installer/pull/7839/commits/b284ecd2236dc15c6dbcb5284b9f542620afe0c9), that allows us to generate CAPI assets as part of the `create manifests` step. There are no tests here for this file yet, but let me see if I can add some.

Reply:
> Nit, typically you make this a pointer since `runtime.Object` and various other interfaces expect this to be a pointer, not sure it matters here so long as the GVK is being set.

Done.

Reply:
> I can either split this out into a private function we can test separately or we can rely on my manual testing to prove this works.

Decision for the installer maintainers if they want this tested or not, but thanks for the demo commit, helpful!